src/spikeinterface/core/template.py

```diff
@@ -0,0 +1,60 @@
+import json
+from dataclasses import dataclass, field
+
+import numpy as np
+
+from spikeinterface.core.sparsity import ChannelSparsity
+
+
+@dataclass
+class Templates:
+    templates_array: np.ndarray
+    sparsity: ChannelSparsity = None
+    num_units: int = field(init=False)
+    num_samples: int = field(init=False)
+    num_channels: int = field(init=False)
+
+    def __post_init__(self):
+        self.num_units, self.num_samples, self.num_channels = self.templates_array.shape
+
+    # Implementing the slicing/indexing behavior as numpy
+    def __getitem__(self, index):
+        return self.templates_array[index]
```

Comment:
It may be better to have a `unit_slice` and `sample/frame_slice` two explicitly slice over the first and last dimension that return a `Templates` object (rather than the sliced `templates_array`).

But maybe in a follow-up PR

Reply:
Yep. If the need arises.